### PR TITLE
Update Thumbnail style

### PR DESCRIPTION
**Pull request by yurm04**
### WHY are these changes introduced?

Fixes #0000 

### How to 🎩

🖥 [Local development instructions](https://github.com/Shopify/polaris/blob/main/README.md#local-development)
🗒 [General tophatting guidelines](https://github.com/Shopify/polaris/blob/main/documentation/Tophatting.md)
📄 [Changelog guidelines](https://github.com/Shopify/polaris/blob/main/.github/CONTRIBUTING.md#changelog)

<details>
<summary>Copy-paste this code in <code>playground/Playground.tsx</code>:</summary>

```jsx
import React from 'react';
import {Page} from '../src';

export function Playground() {
  return (
    <Page title="Playground">
      {/* Add the code you want to test in here */}
    </Page>
  );
}
```

</details>

### 🎩 checklist

- [ ] Tested on [mobile](https://github.com/Shopify/polaris/blob/main/documentation/Tophatting.md#cross-browser-testing)
- [ ] Tested on [multiple browsers](https://help.shopify.com/en/manual/shopify-admin/supported-browsers)
- [ ] Tested for [accessibility](https://github.com/Shopify/polaris/blob/main/documentation/Accessibility%20testing.md)
- [ ] Updated the component's `README.md` with documentation changes
- [ ] [Tophatted documentation](https://github.com/Shopify/polaris/blob/main/documentation/Tophatting%20documentation.md) changes in the style guide

### CODE REVIEW

**Comment by yurm04** [comment]
duplicate PR since @aveline was already working on https://github.com/Shopify/polaris-summer-editions/issues/610

**Comment by sarahill** [comment]
For some reason the inset shadow isn't showing up for me but maybe it's just on my end 🤔 

The code looks good and if I change the box-shadow in the browser it shows up 🤷 

image

**Review by sarahill — APPROVED** [review]
Figured out why the shadow wasn't showing up with Yuraima. Looks good 🎉

File: .changeset/tasty-trees-camp.md [hunk 1]
```diff
@@ -0,0 +1,6 @@
+---
+'@shopify/polaris': minor
+'@shopify/polaris-tokens': minor
+---
+
+Adding border shadow inset and updating thumbnail
```
Comment (laurkim) [inline .changeset/tasty-trees-camp.md]: We haven't been adding changeset entries for polaris-react unless it affects a component without se so I think we can just have this changeset bump `polaris-tokens`.

```suggestion
'@shopify/polaris-tokens': minor
---

Added border shadow inset experimental token
```

File: polaris-tokens/src/token-groups/shadow.ts [hunk 2]
```diff
@@ -125,4 +126,7 @@ export const shadow: {
     value:
       '0px 2px 0px 0px rgba(0, 0, 0, 0.60) inset, 1px 0px 1px 0px rgba(0, 0, 0, 0.20) inset, -1px 0px 1px 0px rgba(0, 0, 0, 0.20) inset',
   },
+  'shadow-border-inset-experimental': {
+    value: 'box-shadow: 0px 0px 0px 1px rgba(0, 0, 0, 0.08) inset',
```
Comment (aaronccasanova) [inline polaris-tokens/src/token-groups/shadow.ts]:
```suggestion
    value: '0px 0px 0px 1px rgba(0, 0, 0, 0.08) inset',
```

File: polaris-react/src/components/Thumbnail/Thumbnail.scss [hunk 3]
```diff
@@ -18,12 +18,32 @@
...
+    &.sizeExtraSmall::after {
+      // stylelint-disable-next-line -- SE23 styling, no 6px radius token
+      border-radius: 6px;
+    }
+
+    &::before {
+      content: '';
+      display: block;
+      // stylelint-disable-next-line -- SE23 styling
+      padding-bottom: 100%;
```
Comment (aaronccasanova) [inline polaris-react/src/components/Thumbnail/Thumbnail.scss:45]: Classic! Perfect use case for the `aspect-ratio` declaration. However, it falls outside our [Safari on iOS](https://caniuse.com/mdn-css_properties_aspect-ratio) browser support requirements:

https://github.com/Shopify/polaris/assets/32409546/4a896f1a-f259-44c9-8d8b-ebc1b68b57f2

**Review by aaronccasanova — APPROVED** [review]

File: polaris-react/src/components/Thumbnail/Thumbnail.scss (hunk 3 above)
Comment (yurm04) [reply polaris-react/src/components/Thumbnail/Thumbnail.scss:45]: @aaronccasanova argh! does this block us merging this in? do we have an established workaround for safari on IOS?

File: polaris-react/src/components/Thumbnail/Thumbnail.scss [hunk 4]
```diff
@@ -18,12 +18,32 @@
...
+      bottom: 0;
+      position: absolute;
+      border-radius: var(--p-border-radius-2);
+      box-shadow: var(--p-shadow-border-inset-experimental);
+    }
+
+    &.sizeExtraSmall,
+    &.sizeExtraSmall::after {
+      // stylelint-disable-next-line -- SE23 styling, no 6px radius token
+      border-radius: 6px;
```
Comment (sarahill) [inline polaris-react/src/components/Thumbnail/Thumbnail.scss:38]: Can we add an experimental border radius token for this? 

Something like: `--p-border-radius-1_5` = `6px`

File: polaris-react/src/components/Thumbnail/Thumbnail.scss (hunk 3 above)
Comment (aaronccasanova) [reply polaris-react/src/components/Thumbnail/Thumbnail.scss:45]: Nope! Your workaround is the pattern I'm familiar with 👍

File: polaris-react/src/components/Thumbnail/Thumbnail.scss (hunk 4 above)
Comment (yurm04) [reply polaris-react/src/components/Thumbnail/Thumbnail.scss:38]: i'll merge in this update and then follow up with another PR

Comment (yurm04) [reply polaris-react/src/components/Thumbnail/Thumbnail.scss:38]: @sarahill [opened PR here](https://github.com/Shopify/polaris/pull/9628)

**Pull request merged**